File: dehaze-go/internal/model/sys_preset.go
package model

import "encoding/json"

type SysPreset struct {
	BaseModel
	Name        string          `gorm:"column:name;type:varchar(64);not null;comment:预设名称" json:"name"`
	Type        string          `gorm:"column:type;type:varchar(16);not null;default:custom;comment:预设类型(system:系统预设;custom:用户自定义)" json:"type"`
	AlgorithmID int64           `gorm:"column:algorithm_id;type:bigint;not null;comment:关联算法ID" json:"algorithmId"`
	Params      *json.RawMessage `gorm:"column:params;type:json;comment:参数键值对(JSON)" json:"params"`
	UserID      *int64          `gorm:"column:user_id;type:bigint;comment:所属用户ID(系统预设为空)" json:"userId"`
	IsDefault   int8            `gorm:"column:is_default;type:tinyint;not null;default:0;comment:是否默认预设" json:"isDefault"`
}

func (SysPreset) TableName() string {
	return "sys_preset"
}

File: dehaze-python/app/models/entity/sys_preset.py
"""参数预设实体"""
from sqlalchemy import BigInteger, Integer, JSON, String
from sqlalchemy.dialects import mysql as mysql_types
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel


class SysPreset(BaseModel):
    __tablename__ = 'sys_preset'
    __table_args__ = {'comment': '参数预设表'}

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True, comment='主键')
    name: Mapped[str] = mapped_column(String(64), nullable=False, comment='预设名称')
    type: Mapped[str] = mapped_column(String(16), nullable=False, default='custom', comment='预设类型(system:系统预设;custom:用户自定义)')
    algorithm_id: Mapped[int] = mapped_column(BigInteger, nullable=False, comment='关联算法ID')
    params: Mapped[dict | None] = mapped_column(JSON, nullable=True, comment='参数键值对(JSON)')
    user_id: Mapped[int | None] = mapped_column(BigInteger, nullable=True, comment='所属用户ID(系统预设为空)')
    is_default: Mapped[int] = mapped_column(mysql_types.TINYINT, nullable=False, default=0, comment='是否默认预设(0:否;1:是)')
